src/events.cpp

```diff
@@ -692,14 +694,14 @@ void onLook(Player* player, const Position& position, Thing* thing, uint8_t stac
 	lua_State* L = scriptInterface.getLuaState();
 	scriptInterface.pushFunction(playerHandlers.onLook);
 
-	tfs::lua::pushUserdata(L, player);
+	tfs::lua::pushSharedPtr(L, player);
 	tfs::lua::setMetatable(L, -1, "Player");
 
-	if (Creature* creature = thing->getCreature()) {
-		tfs::lua::pushUserdata(L, creature);
+	if (auto creature = thing->getCreature()) {
+		tfs::lua::pushSharedPtr(L, creature);
 		tfs::lua::setCreatureMetatable(L, -1, creature);
-	} else if (Item* item = thing->getItem()) {
-		tfs::lua::pushUserdata(L, item);
+	} else if (auto item = thing->getItem()) {
```

Comment:
Could this be `const auto`
```suggestion
	} else if (const auto item = thing->getItem()) {
```